File: test/benchmark/for.lua
local start = os.clock()
for i = 0, 9 do
  local list = {}
  for j = 0, 1999999 do
    list[j] = j
  end

  local sum = 0
  for k, j in pairs(list) do
    sum = sum + j
  end
  io.write(sum .. "\n")
end
io.write(string.format("elapsed: %.8f\n", os.clock() - start))

File: test/benchmark/for.py
from __future__ import print_function

import time

# Map "range" to an efficient range in both Python 2 and 3.
try:
    range = xrange
except NameError:
    pass

start = time.process_time()
for i in range(0, 10):
  list = []
  for j in range(0, 2000000):
    list.append(j)

  sum = 0
  for j in list:
    sum += j
  print(sum)
print("elapsed: " + str(time.process_time() - start))